cli/compose/types/types.go

```diff
@@ -409,19 +562,27 @@ func (e External) MarshalYAML() (interface{}, error) {
 	return External{Name: e.Name}, nil
 }
 
+// MarshalJSON makes External implement json.Marshaller
+func (e External) MarshalJSON() ([]byte, error) {
```

Comment:
Same comment as UlimitsConfig for the `UnmarshalJSON`.

Reply:
Same here
```go

func (e External) MarshalJSON() ([]byte, error) {
	if e.Name == "" {
		return json.Marshal(e.External)
	}
	return json.Marshal(map[string]string{"name": e.Name})
}
```

Reply:
My hunch is that that's probably slower (more function calls), but if you think that's better I'm happy to change it.

Reply:
so I wouldn't worry about *slow* for now, but… it's not a blocker as of now

Reply:
Would there be corner-cases where `e.Name` or `e.External` has to be encoded for JSON?

I agree that performance should not be a huge concern in this case; it's most likely gonna be used interactively, not in a performance-critical path.

Reply:
@thaJeztah If I understand your question correctly, both are simple types (boolean/string), so there's no special casing necessary.